### PR TITLE
Support conditional creation

### PR DESCRIPTION
**Pull request by Necromancerx**
Update:
* support `create_user_pool` for conditional creation of cognito user pool
* remove `prevent_destroy = true` incase user want to delete their cognito user pool via `create_user_pool = false`
* documentation/example added for `create_user_pool` variable

### CODE REVIEW

**Comment by lgallard** [comment]
@Necromancerx thanks for the PR. Just a comment, Terraform 14 lets you use count inside module blocks. So this change would aim to versions 12 & 13

File: README.md [hunk 1]
```diff
@@ -27,6 +27,22 @@ module "aws_cognito_user_pool_simple" {
...
+Sometimes you need to have a way to create Cognito User Pool resources conditionally but Terraform does not allow to use `count` inside `module` block, so the solution is to specify argument `create_user_pool`.
+
+```
+# This Cognito User Pool will not be created
+module "aws_cognito_user_pool_conditional_creation" {
+
+  source  = "lgallard/cognito-user-pool/aws"
+
+  user_pool_name = "conditional_user_pool"
+  create_user_pool = false
```
Comment (lgallard) [inline README.md:41]: Can `create_user_pool` be renamed to `enabled` ? This is more "standard"

**Review by lgallard — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Necromancerx) [reply README.md:41]: That ok. I just based that variable on the VPC module https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws/latest#conditional-creation. I'll update it right away.

**Comment by Necromancerx** [comment]
> @Necromancerx thanks for the PR. Just a comment, Terraform 14 lets you use count inside module blocks. So this change would aim to versions 12 & 13

I didn't know that they already support module blocks on Terraform 14. Thank you!

**Pull request merged**